pkg/scheduler/framework/plugins/nodevolumelimits/non_csi.go

```diff
@@ -0,0 +1,527 @@
+//
+// DEPRECATED
+// All cloudprovider specific predicates defined here are deprecated in favour of CSI volume limit
+// predicate - MaxCSIVolumeCountPred.
+//
+// The predicate looks for both volumes used directly, as well as PVC volumes that are backed by relevant volume
+// types, counts the number of unique volumes, and rejects the new pod if it would place the total count over
+// the maximum.
+func newNonCSILimits(
+	filterName string,
+	csiNodeLister storagelisters.CSINodeLister,
+	scLister storagelisters.StorageClassLister,
+	pvLister corelisters.PersistentVolumeLister,
+	pvcLister corelisters.PersistentVolumeClaimLister,
+) framework.Plugin {
+	var filter VolumeFilter
+	var volumeLimitKey v1.ResourceName
+	var name string
+
+	switch filterName {
+
+	case ebsVolumeFilterType:
+		name = EBSName
+		filter = ebsVolumeFilter
+		volumeLimitKey = v1.ResourceName(volumeutil.EBSVolumeLimitKey)
+	case gcePDVolumeFilterType:
+		name = GCEPDName
+		filter = gcePDVolumeFilter
+		volumeLimitKey = v1.ResourceName(volumeutil.GCEVolumeLimitKey)
+	case azureDiskVolumeFilterType:
+		name = AzureDiskName
+		filter = azureDiskVolumeFilter
+		volumeLimitKey = v1.ResourceName(volumeutil.AzureVolumeLimitKey)
+	case cinderVolumeFilterType:
+		name = CinderName
+		filter = cinderVolumeFilter
+		volumeLimitKey = v1.ResourceName(volumeutil.CinderVolumeLimitKey)
+	default:
+		klog.Fatalf("Wrong filterName, Only Support %v %v %v %v", ebsVolumeFilterType,
+			gcePDVolumeFilterType, azureDiskVolumeFilterType, cinderVolumeFilterType)
+		return nil
+
+	}
+	pl := &nonCSILimits{
+		name:                 name,
+		filter:               filter,
+		volumeLimitKey:       volumeLimitKey,
+		maxVolumeFunc:        getMaxVolumeFunc(filterName),
+		csiNodeLister:        csiNodeLister,
+		pvLister:             pvLister,
+		pvcLister:            pvcLister,
+		scLister:             scLister,
+		randomVolumeIDPrefix: rand.String(32),
+	}
+
+	return pl
+}
+
+// Name returns name of the plugin. It is used in logs, etc.
+func (pl *nonCSILimits) Name() string {
+	return pl.name
+}
+
+// Filter invoked at the filter extension point.
+func (pl *nonCSILimits) Filter(ctx context.Context, _ *framework.CycleState, pod *v1.Pod, nodeInfo *nodeinfo.NodeInfo) *framework.Status {
+	// If a pod doesn't have any volume attached to it, the predicate will always be true.
+	// Thus we make a fast path for it, to avoid unnecessary computations in this case.
+	if len(pod.Spec.Volumes) == 0 {
+		return nil
+	}
+
+	newVolumes := make(map[string]bool)
+	if err := pl.filterVolumes(pod.Spec.Volumes, pod.Namespace, newVolumes); err != nil {
+		return framework.NewStatus(framework.Error, err.Error())
+	}
+
+	// quick return
+	if len(newVolumes) == 0 {
+		return nil
+	}
+
+	node := nodeInfo.Node()
+	if node == nil {
+		return framework.NewStatus(framework.Error, fmt.Sprintf("node not found"))
+	}
+
+	var (
+		csiNode *storage.CSINode
+		err     error
+	)
```

Comment:
not sure this is easier to read than having the var for each variable, this is basically doubling the number of lines used for no reason.

Reply:
done, this is a direct copy of the original code, and I prefer having the var for each of them too,